src/resolve/resolved-dns-scope.c

```diff
@@ -306,55 +306,60 @@ static int dns_scope_socket(
                 int family,
                 const union in_addr_union *address,
                 DnsServer *server,
-                uint16_t port) {
+                uint16_t port,
+                union sockaddr_union **socket_address) {
```

Comment:
for new code we try to follow the rule that output parameters are prefixed with `ret_`, i.e. please rename `socket_address` to `ret_socket_address` here